File: src/main/java/io/scriptor/http/HTTPResultStream.java
package io.scriptor.http;

import java.io.InputStream;

public class HTTPResultStream extends HTTPResultBase<InputStream> {

    public HTTPResultStream(final int statusCode) {
        super(statusCode);
    }

    public HTTPResultStream(final int statusCode, final String statusText) {
        super(statusCode, statusText);
    }

    public HTTPResultStream(final int statusCode, final String statusText, final InputStream value) {
        super(statusCode, statusText, value);
    }

    @Override
    public int getSize() {
        return -1;
    }

    @Override
    public InputStream getStream() {
        return getValue();
    }
}

File: server/src/main/kotlin/dev/scriptor/server/http/result/HTTPResultStream.kt
package dev.scriptor.server.http.result

import dev.scriptor.server.ParameterList
import java.io.InputStream
import java.nio.channels.Channels

class HTTPResultStream : HTTPResult<InputStream> {

    constructor(
        statusCode: Int = 200,
        statusText: String = "OK",
        headers: ParameterList = ParameterList(),
        value: InputStream? = null,
        count: Long = -1L,
    ) : super(
        statusCode,
        statusText,
        headers,
        if (value != null)
            Channels.newChannel(value)
        else null,
        0L,
        count,
    )
}
